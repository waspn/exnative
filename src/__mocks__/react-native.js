jest.mock('react-native', () => {
  const { View } = require.requireActual('react-native')
  return { 
    View: 'View', 
    Text: 'Text', 
    Button: View,
    TextInput: 'TextInput',
    StyleSheet: {
      create: jest.fn(({}) => {
        return {

        }
      }) 
    }
  }
})

//const { View, Text, Button, TextInput, StyleSheet } = require.requireActual('react-native')
//export const View = require.requireActual('react-native')
