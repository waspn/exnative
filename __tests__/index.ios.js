import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

// Note: test renderer must be required after react-native.


const STATUS = {
  HOVERED: 'hovered',
  NORMAL: 'normal',
};

export default class Link extends React.Component {
  constructor() {
    super();

    this._onMouseEnter = this._onMouseEnter.bind(this);
    this._onMouseLeave = this._onMouseLeave.bind(this);

    this.state = {
      class: STATUS.NORMAL,
    };
  }

  _onMouseEnter() {
    this.setState({class: STATUS.HOVERED});
  }

  _onMouseLeave() {
    this.setState({class: STATUS.NORMAL});
  }

  render() {
    return (
      <a
        className={this.state.class}
        href={this.props.page || '#'}
        onMouseEnter={this._onMouseEnter}
        onMouseLeave={this._onMouseLeave}
      >
        {this.props.children}
      </a>
    );
  }
}


it('renders correctly', () => {
  const tree = renderer.create(
      <Link page="http://www.google.com">Google</Link>
  ).toJSON();
  expect(tree).toMatchSnapshot();
})