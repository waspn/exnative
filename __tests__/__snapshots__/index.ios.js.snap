// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="normal"
  href="http://www.google.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Google
</a>
`;
